src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -172,26 +175,30 @@ public void start(final Map<String, String> props) {
         return sourceRecords.isEmpty() ? null : sourceRecords;
       } else {
         BsonDocument changeStreamDocument = next.get();
-
         Map<String, String> sourceOffset = new HashMap<>();
         sourceOffset.put("_id", changeStreamDocument.getDocument("_id").toJson());
         if (isCopying.get()) {
           sourceOffset.put("copy", "true");
         }
 
+
         String topicName =
             getTopicNameFromNamespace(
                 prefix, changeStreamDocument.getDocument("ns", new BsonDocument()));
 
-        Optional<String> jsonDocument = Optional.empty();
+        Optional<BsonDocument> bsonDocument = Optional.empty();
         if (publishFullDocumentOnly) {
           if (changeStreamDocument.containsKey("fullDocument")) {
-            jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+              bsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument"));
+
           }
         } else {
-          jsonDocument = Optional.of(changeStreamDocument.toJson());
+
+            bsonDocument = Optional.of(changeStreamDocument);
         }
 
+        Optional<String> jsonDocument;
+        jsonDocument = convertToJson(bsonDocument, jsonOptions);
```

Comment:
You can join the declaration and the assignment:
```suggestion
        Optional<String> jsonDocument = convertToJson(bsonDocument, jsonOptions);
```